src/core/environment.js

```diff
@@ -737,4 +737,80 @@ p5.prototype.getURLParams = function() {
   return v;
 };
 
+/**
+ * Reduces the log load to get the better performance and
+ * more informative logs from console, by ignoring repeated values or setting periods.
+ * Instead of using console.log(a) to display the value of variable a,
+ * now you may use lazyLog('a'). Calling the function without any parameters will
+ * clear the cache of logged variables.
+ *
+ * @method lazyLog
+ * @param {Object} variable can be any JavaScript data type
+ * @param {String} variableName a tag to help cache different variables, can just be the same as the variable name
+ * @param {Number} [period] number of frames (calls of the function) not logging
+ * @param {Boolean} [display] set to false if not displaying the name of variable in log
+ * @return {Boolean} true if the variable changes or match the logging condition, otherwise false
+ * @example
+ * <div>
+ * <code>
+ * // Log only once when the value stays unchanged
+ * let x = 50;
+ * function draw() {
+ *   ellipse(x, 50, 50);
+ *   lazyLog(x, 'x');
+ * }
+ * </code>
+ * </div>
+ *
+ * <div>
+ * <code>
+ * // Log frame rate
+ * function draw() {
+ *   lazyLog(int(getFrameRate()), 'fr');
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * Log only once when the value stays unchanged.
+ * Log frame rate.
+ */
+/**
+ * @method lazyLog
+ */
+p5.prototype.lazyLog = function(...args) {
+  let argsLength = args.length;
+  let displayName = true;
+
+  if (!argsLength) {
+    // Clear the cached variables
+    this._lazyLogVars = {};
+    this._lazyLogTimers = {};
+    return false;
+  }
+  if (typeof args[argsLength - 1] === 'boolean' && !args[argsLength - 1]) {
```

Comment:
i wonder if unpacking the args at entry to `lazyLog` would make it easier to follow how the different cases work. i think i get it, but i would have more confidence making a change to this in the future if the args were unpacked and labeled.